### PR TITLE
Bug 1279042 - show username from prefs as chat displayName

### PR DESCRIPTION
**Pull request by dmose**
r? @fcampo, @mancas, @daveccrit

### CODE REVIEW

File: add-on/panels/js/roomToc.jsx [hunk 1]
```diff
@@ -47,9 +47,16 @@ loop.roomToc = (function(mozL10n) {
...
       let pageStore = new loop.store.PageStore(dispatcher, { dataDriver });
 
-      // XXX akita bug 1279042 Use user set name instead of fake name.
-      dispatcher.dispatch(
-        new sharedActions.SetOwnDisplayName({ displayName: "Room Owner" }));
+      // Load the username from storage, or use random guest name
+      loop.request("GetLoopPref", "username")
+        .then(storedName => {
+          let username = storedName ||
+                         "Room Owner";
```
Comment (mancas) [inline add-on/panels/js/roomToc.jsx]: We should add a comment here, to add the default name to locale files later.

File: shared/js/utils.js [hunk 2]
```diff
@@ -172,6 +172,49 @@ if (inChrome) {
...
+      });
+      return;
+    }
+
+    callback(localStorage.getItem(prefName));
+  }
+
+  /**
+   * Used for setting a String preference. It will either use the browser
+   * preferences, or try to get them from localStorage, depending on whether
```
Comment (mancas) [inline shared/js/utils.js]: nit: try to get them from localStorage <- we're setting the preference

File: shared/js/utils.js [hunk 3]
```diff
@@ -172,6 +172,49 @@ if (inChrome) {
...
+  function setStringPreference(prefName, prefValue, callback) {
+    if (loop.shared.utils.isDesktop()) {
+     loop.request("SetLoopPref", prefName, prefValue).then(function() {
+       callback(String(prefValue));
+     });
+
+     return;
+    }
+    //
+    // callback(String(localStorage.getItem(prefName)));
```
Comment (mancas) [inline shared/js/utils.js]: I'm sure we'll need to set the pref from guest as well.

Comment (mancas) [inline shared/js/utils.js]: nit: localStorage.setItem(prefName, prefValue)

File: shared/test/utils_test.js [hunk 4]
```diff
@@ -122,6 +122,115 @@ describe("loop.shared.utils", function() {
...
+    describe("mozLoop not set", function() {
+      beforeEach(function() {
+        sandbox.stub(loop.shared.utils, "isDesktop").returns(false);
+      });
+
+      it("should set the localStorage value", function() {
+        sharedUtils.setStringPreference("test.someStringPref", "sheep", function() {
+          expect(localStorage.getItem("test.someStringPref")).eql("sheep");
+        });
+      });
```
Comment (mancas) [inline shared/test/utils_test.js]: Not sure why this test is working if the localStorage stuff is commented

File: shared/js/utils.js [hunk 5]
```diff
@@ -172,6 +172,49 @@ if (inChrome) {
...
+  /**
+   * Used for getting a String preference. It will either use the browser
+   * preferences or try to get them from localStorage.
+   *
+   * @param {String}   prefName The name of the preference. Note that mozLoop
+   *                            adds 'loop.' to the start of the string.
+   * @param {Function} callback Invoked with the preference value.  null is
+   *                            returned if no such preference is found.
+   */
+  function getStringPreference(prefName, callback) {
```
Comment (fcampo) [inline shared/js/utils.js]: Not sure we need this function, when we already check from `localStorage` when we use `loop.request("GetLoopPref", pref)` from the standalone side ([standaloneMozLoop](https://github.com/mozilla/loop/blob/master/standalone/content/js/standaloneMozLoop.js#L310-L313))

File: shared/js/utils.js [hunk 6]
```diff
@@ -172,6 +172,49 @@ if (inChrome) {
...
+   * preferences, or try to get them from localStorage, depending on whether
+   * we're on the desktop or not.
+   *
+   * @param {String}   prefName The name of the preference. Note that mozLoop
+   *                            adds 'loop.' to the start of the string.
+   * @param {String}   prefValue
+   * @param {Function} callback Invoked with the preference value for
+   *                            convenience, coerced to String if necessary.
+   */
+  function setStringPreference(prefName, prefValue, callback) {
```
Comment (fcampo) [inline shared/js/utils.js]: [same](https://github.com/mozilla/loop/blob/master/standalone/content/js/standaloneMozLoop.js#L296-L301) here

File: shared/js/utils.js (hunk 5 above)
Comment (dmose) [inline shared/js/utils.js]: Quite right; I hadn't seen that.  I was following the getBoolPreference pattern, but now I sort of assume that the point of getBoolPreference is less about the encapsulation of preferences, and more about the boolean coercion.  Weird.

File: add-on/panels/js/roomToc.jsx (hunk 1 above)
Comment (dmose) [inline add-on/panels/js/roomToc.jsx]: Done.

**Pull request merged**